src/HTTP/Psr7Response.php

```diff
@@ -0,0 +1,199 @@
+     *             header(sprintf('%s: %s', $name, $value), false);
+     *         }
+     *     }
+     *
+     * While header names are not case-sensitive, get_headers() will preserve the
+     * exact case in which headers were originally specified.
+     *
+     * @return string[][] Returns an associative array of the message's headers.
+     *     Each key MUST be a header name, and each value MUST be an array of
+     *     strings for that header.
+     */
+    public function get_headers(): array
+    {
+        return $this->response->getHeaders();
+    }
+
+    /**
+     * Checks if a header exists by the given case-insensitive name.
+     *
+     * @param string $name Case-insensitive header field name.
+     * @return bool Returns true if any header names match the given header
+     *     name using a case-insensitive string comparison. Returns false if
+     *     no matching header name is found in the message.
+     */
+    public function has_header(string $name): bool
+    {
+        return $this->response->hasHeader($name);
+    }
+
+    /**
+     * Retrieves a message header value by the given case-insensitive name.
+     *
+     * This method returns an array of all the header values of the given
+     * case-insensitive header name.
+     *
+     * If the header does not appear in the message, this method MUST return an
+     * empty array.
+     *
+     * @param string $name Case-insensitive header field name.
+     * @return string[] An array of string values as provided for the given
+     *    header. If the header does not appear in the message, this method MUST
+     *    return an empty array.
+     */
+    public function get_header(string $name): array
+    {
+        return $this->response->getHeader($name);
+    }
+
+    /**
+     * Retrieves a comma-separated string of the values for a single header.
+     *
+     * This method returns all of the header values of the given
+     * case-insensitive header name as a string concatenated together using
+     * a comma.
+     *
+     * NOTE: Not all header values may be appropriately represented using
+     * comma concatenation. For such headers, use getHeader() instead
+     * and supply your own delimiter when concatenating.
+     *
+     * If the header does not appear in the message, this method MUST return
+     * an empty string.
+     *
+     * @param string $name Case-insensitive header field name.
+     * @return string A string of values as provided for the given header
+     *    concatenated together using a comma. If the header does not appear in
+     *    the message, this method MUST return an empty string.
+     */
+    public function get_header_line(string $name): string
+    {
+        return $this->response->getHeaderLine($name);
+    }
+
+    /**
+     * get the body as string
+     *
+     * @return string
+     */
+    public function get_body_content(): string
+    {
+        return $this->response->getBody()->getContents();
```

Comment:
We should use `__toString()` and cache it, otherwise repeated calls of `getContents()` might not work: https://www.php-fig.org/psr/psr-7/#34-psrhttpmessagestreaminterface

Reply:
Yes, we should use `__toString()` because `getContents()` returns the remaining contents, but will not seek to the beginning of the stream.